src/DistributedMetadata/CatalogService.cpp

```diff
@@ -305,6 +305,19 @@ std::pair<CatalogService::TablePtr, StoragePtr> CatalogService::findTableStorage
     return {table_p, nullptr};
 }
 
+void CatalogService::deleteTableStorageByName(const String & database, const String & table)
+{
+    std::shared_lock storage_guard{storage_rwlock};
+
+    auto storage_iter = storages.find(std::make_pair(database, table));
+    if (storage_iter != storages.end())
+    {
+        auto removed = storages.erase(storage_iter);
+        assert(removed == 1);
+        (void)removed;
```

Comment:
we can clean up this line as well

Reply:
So the function will be just simplify as 

```
std::shared_lock storage_guard{storage_rwlock};
storages.erase(std::make_pair(database, table));
```

Reply:
fixed